pkg/kubectl/resource_filter.go

```diff
@@ -0,0 +1,110 @@
+package kubectl
+
+import (
+	"fmt"
+	"io"
+	"reflect"
+
+	"github.com/golang/glog"
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/runtime"
+)
+
+type FilterFunc func(runtime.Object) (bool, error)
+
+type FilterOptions struct {
```

Comment:
also, it's weird that this is a public type but only has private members.  If you really want to keep the implementation private, I would add an interface for the public functions and then have the struct implement the interface.